### PR TITLE
Strip

### PR DESCRIPTION
**Pull request by J2thatsme**

### CODE REVIEW

File: types/sensative/strip/alarm.schema.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{
+  "$id": "https://akenza.io/sensative/strip/alarm.schema.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Alarm",
+  "type": "object",
+  "properties": {
+    "highAlarm": {
+      "title": "High alarm",
+      "description": "Equals true if the threshold is exceeded",
```
Comment (mariusgiger) [inline types/sensative/strip/alarm.schema.json:9]: typo Equals **to** true, also change all the others

File: types/sensative/strip/alarm.schema.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      "type": "boolean"
+    },
+    "lowAlarm": {
+      "title": "Low alarm",
+      "description": "Equals true if the threshold is fallen below",
```
Comment (mariusgiger) [inline types/sensative/strip/alarm.schema.json:14]: typo is -> has

File: types/sensative/strip/alarm.schema.json [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      "type": "boolean"
+    },
+    "doorAlarm": {
+      "title": "Door alarm",
+      "description": "Equals true if theres a door alarm",
```
Comment (mariusgiger) [inline types/sensative/strip/alarm.schema.json:19]: typo theres -> there is

File: types/sensative/strip/alarm.schema.json [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      "title": "Flood alarm",
+      "description": "Equals true if a flood is detected",
+      "type": "boolean"
+    },
+    "foilAlarm": {
+      "title": "Foil alarm",
+      "description": "Equals true if a foil is detected",
+      "type": "boolean"
+    },
+    "userSwitch1Alarm": {
```
Comment (mariusgiger) [inline types/sensative/strip/alarm.schema.json:37]: userSwitch1Alarm -> userSwitchAlarm?

File: types/sensative/strip/default.schema.json [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+      "type": "boolean"
+    },
+    "irProximity": {
+      "title": "Infrared proximity",
+      "description": "Count of objects in proximity",
+      "type": "integer",
+      "minimum": 0,
+      "maximum": 255
+    },
+    "irCloseproximity": {
```
Comment (mariusgiger) [inline types/sensative/strip/default.schema.json:84]: change to irCloseProximity

File: types/sensative/strip/uplink.js [hunk 6]
```diff
@@ -0,0 +1,211 @@
+function decoder(bytes, port) {
+  // Decode an uplink message from a buffer
+  // (array) of bytes to an object of fields.
+
+  function decodeFrame(type, target) {
```
Comment (mariusgiger) [inline types/sensative/strip/uplink.js:5]: why function in function?

File: types/sensative/strip/uplink.js [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+  var port = event.data.port;
+  var data = decoder(parseHexString(payload), port);
+
+  var lifecycle = {}; var alarm = {}; var def = {};
+  var lifecycleOut = false; var alarmOut = false; var defOut = false;
+  var defaultKeys = ["temperature", "averageTemperature", "humidity", "lux", "lux2", "open", "tamperReport", "flood", "doorCount", "presence", "irProximity", "irCloseproximity"];
+  var alarmKeys = ["highAlarm", "lowAlarm", "doorAlarm", "tamperAlarm", "floodAlarm", "foilAlarm", "userSwitch1Alarm", "closeProximityAlarm", "disinfectAlarm"];
+  var lifecycleKeys = ["statusPercent", "error", "historySeqNr", "prevHistSeqNr"];
+
+  Object.keys(data).forEach(function (key) {
```
Comment (mariusgiger) [inline types/sensative/strip/uplink.js:179]: not sure if this is a good way of handling that, better make it explicit

File: types/sensative/strip/uplink.spec.js [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+        if (value.topic === "default") {
+
+          assert.equal(value.data.temperature, 0);
+          assert.equal(value.data.open, true);
+
+          validate(value.data, defaultSchema, { throwError: true });
+        }
+
+        if (value.topic === "alarm") {
+
```
Comment (mariusgiger) [inline types/sensative/strip/uplink.spec.js:87]: remove unused new lines

**Review by mariusgiger — CHANGES_REQUESTED** [review]

File: types/sensative/strip/uplink.js (hunk 7 above)
Comment (J2thatsme) [reply types/sensative/strip/uplink.js]: do we want to handle it the same as decentlab ?

**Review by mariusgiger — APPROVED** [review]

**Pull request merged**